File: src/main/java/com/demo/report/constants/ReportTypes.java
package com.demo.report.constants;

import java.util.List;

public class ReportTypes {

    public static final String FMT_PDF = "PDF";
    public static final String FMT_EXCEL = "EXCEL";

    // EMPLOYEE(USER)
    public static final String PERSONAL_DETAIL_EXCEL = "PERSONAL_DETAIL_EXCEL";
    public static final String PERSONAL_SUMMARY_PDF  = "PERSONAL_SUMMARY_PDF";

    // ADMIN
    public static final String DEPT_DETAIL_EXCEL     = "DEPT_DETAIL_EXCEL";
    public static final String DEPT_SUMMARY_PDF      = "DEPT_SUMMARY_PDF";
    public static final String AI_STRATEGY_PDF       = "AI_STRATEGY_PDF";

    public record TypeDef(String id, String label, String format, boolean adminOnly) {}

    public static final List<TypeDef> ALL = List.of(
            new TypeDef(PERSONAL_DETAIL_EXCEL, "Personal Detailed Records (Excel)", FMT_EXCEL, false),
            new TypeDef(PERSONAL_SUMMARY_PDF,  "Personal Summary Report (PDF)",     FMT_PDF,   false),

            new TypeDef(DEPT_DETAIL_EXCEL,     "Department Detailed Records (Excel)", FMT_EXCEL, true),
            new TypeDef(DEPT_SUMMARY_PDF,      "Department Summary Report (PDF)",     FMT_PDF,   true),
            new TypeDef(AI_STRATEGY_PDF,       "AI Strategy Insight Report (PDF)",    FMT_PDF,   true)
    );

    public static TypeDef find(String id) {
        return ALL.stream().filter(t -> t.id().equals(id)).findFirst().orElse(null);
    }
}

File: team1/src/constants/reportTypes.js
// src/constants/reportTypes.js
export const REPORT_TYPES = [
    // ===== EMPLOYEE =====
    {
        id: "personal_detail_excel",
        label: "Personal Detailed Records (Excel)",
        outputFormat: "Excel",
        roles: ["EMPLOYEE"],
        preview: { recordsIncluded: 42, totalAmount: "₩1,240,000" },
    },
    {
        id: "personal_summary_pdf",
        label: "Personal Summary Report (PDF)",
        outputFormat: "PDF",
        roles: ["EMPLOYEE"],
        preview: { recordsIncluded: 42, totalAmount: "₩1,240,000" },
    },

    // ===== ADMIN =====
    {
        id: "dept_detail_excel",
        label: "Department Detailed Records (Excel)",
        outputFormat: "Excel",
        roles: ["ADMIN"],
        preview: { recordsIncluded: 128, totalAmount: "₩7,920,000" },
    },
    {
        id: "dept_summary_pdf",
        label: "Department Summary Report (PDF)",
        outputFormat: "PDF",
        roles: ["ADMIN"],
        preview: { recordsIncluded: 128, totalAmount: "₩7,920,000" },
    },
    {
        id: "ai_strategy_pdf",
        label: "AI Strategy Insight Report (PDF)",
        outputFormat: "PDF",
        roles: ["ADMIN"],
        preview: { recordsIncluded: 128, totalAmount: "₩7,920,000" },
    },
];
